src/api/Feed.js

```diff
@@ -165,27 +166,38 @@ class Feed extends Base {
     updateAnnotation = (
         file: BoxItem,
         annotationId: string,
-        text: string,
+        text?: string,
+        status?: FeedItemStatus,
         permissions: AnnotationPermission,
         successCallback: (annotation: Annotation) => void,
         errorCallback: ErrorCallback,
     ): void => {
         if (!file.id) {
             throw getBadItemError();
         }
+        if (!text && !status) {
+            throw new Error('Missing text or status!');
+        }
 
         this.annotationsAPI = new AnnotationsAPI(this.options);
 
         this.file = file;
         this.errorCallback = errorCallback;
 
-        this.updateFeedItem({ message: text, isPending: true }, annotationId);
+        const feedItemChanges = {};
+        if (text) {
```

Comment:
Do we need these checks at this layer or can we defer validation to the API, as we do now?

Reply:
I believe this is needed because `this.updateFeedItem` is called below which updates the annotation temporarily (when the api request is not yet resolved) overwriting it's properties with the properties of the passed `updates` object. Before only message could be updated so the new message was displayed along with the loading indicator. Now when only `status` could be changed then `message` would be `undefined` so during this "loading time" the text would disappear.

Reply:
Got it, thanks for clarifying.